mode/smalltalk/smalltalk.js

```diff
@@ -91,6 +92,7 @@ CodeMirror.defineMode('smalltalk', function(config) {
     } else if (/[\w_]/.test(aChar)) {
       stream.eatWhile(/[\w\d_]/);
       token.name = state.expectVariable ? (keywords.test(stream.current()) ? 'keyword' : 'variable') : null;
+      token.name = state.expectVariable ? (variables.test(stream.current()) ? 'variable-2' : 'variable') : null;
```

Comment:
Why are you assigning to the same property twice here? This effectively disables the previous statement, which doesn't look like it is right.